Source/JavaScriptCore/runtime/StackFrame.cpp

```diff
@@ -58,16 +58,22 @@ SourceID StackFrame::sourceID() const
     return m_codeBlock->ownerExecutable()->sourceID();
 }
 
-String StackFrame::sourceURL() const
+String StackFrame::sourceURL(VM& vm) const
 {
     if (m_isWasmFrame)
         return "[wasm code]"_s;
 
-    if (!m_codeBlock) {
+    if (!m_codeBlock)
         return "[native code]"_s;
-    }
 
     String sourceURL = m_codeBlock->ownerExecutable()->sourceURL();
+
+    if (vm.clientData) {
+        String overrideURL = vm.clientData->overrideSourceURL(*this, sourceURL);
```

Comment:
Can you filter out http/https URLs to avoid calling a virtual function here?
Right now, only WebKit/WebCore is implementing this interface. And it will limit `http`/`https`. So I think we can first filter out them here. If we have any other motivation further filtering these schemes even in JavaScriptCore.framework, I think we can revisit and extend it later.

Reply:
Are you okay doing this (since I wanted to avoid passing the and making a `URL` in JSC)?

```C++
    if (vm.clientData && !sourceURL.startsWithIgnoringASCIICase("http")) {
```

Reply:
No need, `URL.h` has a `protocolIsInHTTPFamily` function which takes a `StringView`, not a `URL`.

Reply:
Hope I didn’t catch you too late!

Reply:
Could you do a follow-up patch to remove the explicit `"http"_s`?